### PR TITLE
Sprucing up the report

### PR DESCRIPTION
**Pull request by standage**
There have been several updates to the report recently, focused primarily on improving visualizations. This PR adds figure numbers & captions, table numbers & titles, and updates the text in a few places where it has become redundant, verbose, or outdated.

----------

- [x] Changes are clearly described above
- [x] Any relevant issue threads are referenced in the description
- [x] Any new features are tested (see the [development manual](https://microhapulator.readthedocs.io/en/stable/devel.html) for details)
- [x] CLI documentation (see [docs/cli.md](docs/cli.md)) and Python API documentation (see [microhapulator/api.py](microhapulator/api.py)) are up-to-date and in sync
- [x] Substantial changes are documented in CHANGELOG.md (see https://keepachangelog.com/en/1.0.0/)

### CODE REVIEW

**Review by agshumate — COMMENTED** [review]
These updates look really nice. I especially like the figure tables and captions. It looks like you did the same re-ordering that I did in #173, so the only other difference is the read length filtering tables. If we add those to this branch, that should give us the desired report, and then we can just accept all incoming changes from this branch when merging into master and resolving conflicts. Let me know if you think there is a better strategy!

**Comment by standage** [comment]
I merged "main" and resolved the conflicts, adding the length filtering table results as we had discussed. This works well for the paired end data, where ambiguity filtering and length filtering are split by the read merging step. But for single end data, putting these tables right after each other in the report didn't work so well. So I merged them into a single table.

image

**Comment by standage** [comment]
The template code has gotten messy, and I think we should do a bit more preprocessing in Python to collate the data so we don't have to do too much processing directly in the Jinja template.

**Comment by standage** [comment]
> The template code has gotten messy, and I think we should do a bit more preprocessing in Python to collate the data so we don't have to do too much processing directly in the Jinja template.

Looking at the relevant code, I think we should handle this in a dedicated branch. Maybe a pair programming session at some point?

File: microhapulator/data/template.html [hunk 1]
```diff
@@ -1,328 +1,297 @@
...
-            chi-square statistic is a measure of read coverage imbalance between markers, and can be compared among
-            samples sequenced using the sample panel: the minimum value of 0 represents perfectly uniform coverage
-            across markers, while the maximum value of D occurs when all reads map to a single marker (D represents the
-            degrees of freedom, or the number of markers minus 1).
-        </p>
-        <div class="scrollwrapper">
+            <a name="readmapping"></a>
+            <h2>Read Mapping</h2>
+            <p>Filtered and merged reads are aligned to marker reference sequences using <a href="http://bio-bwa.sourceforge.net/bwa.shtml" target="_blank">BWA MEM</a> and formatted, sorted, and indexed using <a href="http://www.htslib.org/" target="_blank">SAMtools</a>.</p>
+            <p class="title"><strong>Table 3.1</strong>: Read mapping metrics.</p>
```
Comment (agshumate) [inline microhapulator/data/template.html]: "Filtered and _merged_ reads..."  appears for single ended data.

**Review by agshumate — COMMENTED** [review]
Made one very minor comment but overall it looks good! I agree that doing some pair programming to make the template cleaner would be helpful. It is getting kind of out of control.

File: microhapulator/data/template.html (hunk 1 above)
Comment (standage) [reply microhapulator/data/template.html]: Good catch. Handled this with an inline conditional.

```diff
@@ -280,7 +280,7 @@
 
             <a name="readmapping"></a>
             <h2>Read Mapping</h2>
-            <p>Filtered and merged reads are aligned to marker reference sequences using <a href="http://bio-bwa.sourceforge.net/bwa.shtml" target="_blank">BWA MEM</a> and formatted, sorted, and indexed using <a href="http://www.htslib.org/" target="_blank">SAMtools</a>.</p>
+            <p>Filtered {% if reads_are_paired %} and merged {% endif %} reads are aligned to marker reference sequences using <a href="http://bio-bwa.sourceforge.net/bwa.shtml" target="_blank">BWA MEM</a> and formatted, sorted, and indexed using <a href="http://www.htslib.org/" target="_blank">SAMtools</a>.</p>
             <p class="title"><strong>Table 3.1</strong>: Read mapping metrics.</p>
             <table>
                 <tr>
```

**Pull request merged**